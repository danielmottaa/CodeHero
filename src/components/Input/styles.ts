import styled from 'styled-components/native';

export const Container = styled.View``;

export const InputCustom = styled.TextInput`
  border: 1px solid ${({ theme }) => theme.Colors?.DIM_GRAY};
  height: 30px;
  border-radius: 5px;
  margin-top: 3px;
  padding: 4px;
`;
