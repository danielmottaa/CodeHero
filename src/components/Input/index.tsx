import React from 'react';
import Colors from '../../global/colors';
import { IInput } from './Models/indext';

import * as S from './styles';

const Input: React.FC<IInput> = ({
  placeholder,
  value,
  onChangeText
}) => {
  return (
    <S.Container>
      <S.InputCustom
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
        placeholderTextColor={Colors.DIM_GRAY}
      />
    </S.Container>
  );
}

export default Input;