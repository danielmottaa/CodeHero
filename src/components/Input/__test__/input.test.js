import React from 'react';
import renderer from 'react-test-renderer';

import Input from '..';

describe('<Input />', () => {
  it('rating was passed for input', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Input />);
    expect(tree).toMatchSnapshot();
  });
});
