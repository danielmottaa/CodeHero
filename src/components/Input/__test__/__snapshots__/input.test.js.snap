// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<View
  style={
    [
      {},
    ]
  }
>
  <TextInput
    placeholderTextColor="#696969"
    style={
      [
        {
          "borderColor": "black",
          "borderRadius": 5,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 30,
          "marginTop": 3,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    testID="input"
  />
</View>
`;
